server.py

```
@@ -55,23 +55,335 @@
+                    # Pass the port this process is actually bound to so the
+                    # LaunchAgent reboots on the right port, not the 7870
+                    # default. Operators frequently pick a custom port when
+                    # another agent is already on 7870.
+                    ok_as, msg_as = install_autostart(agent_name=as_name, port=_active_port)
+                else:
+                    ok_as, msg_as = uninstall_autostart(agent_name=as_name)
+                messages.append(f"autostart: {msg_as}")
+                if not ok_as:
+                    log.warning("[setup] autostart sync failed: %s", msg_as)
+            except Exception as e:
+                log.exception("[setup] autostart sync raised: %s", e)
```

Ruff (0.15.10)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)